src/deluge/gui/views/automation_instrument_clip_view.cpp

```diff
@@ -753,21 +733,14 @@ void AutomationInstrumentClipView::renderAutomationEditor(ModelStackWithTimeline
 //this function started off as a copy of the renderRow function from the NoteRow class - I replaced "notes" with "nodes"
 //it worked for the most part, but there was bugs so I removed the buggy code and inserted my alternative rendering method
 //which always works. hoping to bring back the other code once I've worked out the bugs.
-void AutomationInstrumentClipView::renderRow(ModelStackWithAutoParam* modelStack, uint8_t* image,
-                                             uint8_t occupancyMask[], bool overwriteExisting,
-                                             uint32_t effectiveRowLength, bool allowNoteTails, int32_t xScroll,
-                                             uint32_t xZoom, int32_t xStartNow, int32_t xEnd, bool drawRepeats,
-                                             int32_t yDisplay, bool isAutomated) {
+void AutomationInstrumentClipView::renderRow(ModelStackWithTimelineCounter* modelStack,
+                                             ModelStackWithAutoParam* modelStackWithParam, uint8_t* image,
+                                             uint8_t occupancyMask[], int32_t yDisplay, bool isAutomated) {
 
 	for (int32_t xDisplay = 0; xDisplay < kDisplayWidth; xDisplay++) {
 
```

Comment:
refactored code for getting middle position of a pad out to getMiddlePosFromSquare